render/vulkan/shaders/meson.build

```diff
@@ -0,0 +1,20 @@
+vulkan_shaders_src = [
+	'common.vert',
+	'texture.frag',
+	'quad.frag',
+]
+
+vulkan_shaders = []
+foreach shader : vulkan_shaders_src
+	name = shader.underscorify() + '_data'
+	args = [glslang, '-V', '@INPUT@', '-o', '@OUTPUT@', '--vn', name]
```

Comment:
You should add the '-Os' parameter too. By default, glslang does very few optimizations, if any (I've never seen it do any without -Os). They currently have no other optimization options, other than for size, which does make complex shaders around 10%-20% faster in the best case, which does make a difference.

Reply:
I doubt that it'll have a big impact for our shaders but I agree, we should do it.

Reply:
Could be conditional on the Meson build type.

Reply:
I noticed just now that some distributions (e.g. Arch) ship glslangValidator without the optimization feature, `-Os` causes an error then. We could require spirv-opt (spirv-tools package on Arch) to be installed and use that manually (and then use yet another pass to transform that spirv into a C array header) but IMO the complexity is worth it right now. I compared the optimized texture.frag spirv to the unoptimized one and the only difference seems to be the removal of duplicate OpAccessChain declarations which should be irrelevant for shader performance. We can look into this if we ever have more complicated shaders.

Reply:
Sounds good!